homeassistant/components/backup/config.py

```diff
@@ -567,6 +561,35 @@ def _delete_filter(
     )
 
 
+@callback
+def check_unavailable_agents(
+    hass: HomeAssistant, manager: BackupManager, new_agent_ids: list[str] | None = None
+) -> None:
+    """Check for unavailable agents."""
+    if missing_agent_ids := set(manager.config.data.create_backup.agent_ids) - set(
+        manager.backup_agents
+    ):
+        LOGGER.debug(
+            "Agents %s are configured for automatic backup but are not loaded",
+            missing_agent_ids,
+        )
+        for agent_id in missing_agent_ids:
+            create_automatic_backup_agents_unavailable_issue(hass, agent_id)
+
+    # Remove any issues for agents that are now loaded
+    for agent_id in manager.backup_agents:
+        delete_automatic_backup_agents_unavailable_issue(hass, agent_id)
+
+    if new_agent_ids is None:
+        return
+
+    # Remove issues for agents that are no longer configured
+    for agent_id in set(manager.config.data.create_backup.agent_ids) - set(
+        new_agent_ids
+    ):
+        delete_automatic_backup_agents_unavailable_issue(hass, agent_id)
```

Comment:
This seems error prone, there's a risk we don't clean up issues.
We should instead make a collection of issue registry issues for missing agents, and remove all which are not in the `missing_agent_ids` set above.

This function does something similar: https://github.com/home-assistant/core/blob/6d80ccaeb110ad31f46a051e24590233d7be3365/homeassistant/components/sensor/recorder.py#L752-L766

Reply:
Thanks! I think I've addressed this now and it made everything a lot better with less code.